changelog.d/14817.bugfix

```diff
@@ -0,0 +1 @@
+Fix race where calling `/members` or `/state` with an `at` parameter could fail for newly created rooms, when using multiple workers.
```

Comment:
Is it just newly-created rooms that are vulnerable to this race? E.g. are there other consumers of `get_last_event_in_room_before_stream_ordering`?

Reply:
Hmm, it can also be a bit weird if it races with any state change, as it may return the state *before* the state change. And I guess joins would also do something weird here.

Will update.